numba/tests/test_npdatetime.py

```diff
@@ -903,6 +939,81 @@ def test_add_td_no_match(self):
     def test_sub_td_no_match(self):
         self._test_add_sub_td_no_match(np.subtract)
 
+    def _test_min_max(self, operation, parallel, method):
+        if method:
+            if operation is np.min:
+                def impl(arr):
+                    return arr.min()
+            else:
+                def impl(arr):
+                    return arr.max()
+        else:
+            def impl(arr):
+                return operation(arr)
+
+        py_func = impl
+        cfunc = njit(parallel=parallel)(impl)
+
+        test_cases = [
+            np.array([
+                DT("2011-01-01", "ns"),
+                DT("1971-02-02", "ns"),
+                DT("2021-03-03", "ns"),
+                DT("2004-12-07", "ns"),
+            ]),
+            np.array([
+                DT("2011-01-01", "D"),
+                DT("1971-02-02", "D"),
+                DT("2021-03-03", "D"),
+                DT("2004-12-07", "D"),
+            ]),
+            np.array([
+                TD(1, "ns"),
+                TD(2, "ns"),
+                TD(3, "ns"),
+                TD(4, "ns"),
+            ]),
+            np.array([
+                TD(1, "D"),
+                TD(2, "D"),
+                TD(3, "D"),
+                TD(4, "D"),
+            ]),
+        ]
+
+        for arr in test_cases:
```

Comment:
Can you add a test with NaT in each array type? In addition can you add a test for the edge case where the data is all NaT.